pkg/controller/node/status.go

```diff
@@ -191,9 +192,14 @@ func getUnavailableMachines(currentConfig string, nodes []*corev1.Node) []*corev
 		if !ok || cconfig == "" {
 			continue
 		}
+		dstate, ok := node.Annotations[daemonconsts.MachineConfigDaemonStateAnnotationKey]
+		if !ok || dstate == "" {
+			continue
+		}
 
 		nodeNotReady := !isNodeReady(node)
-		if dconfig == currentConfig && (dconfig != cconfig || nodeNotReady) {
+		// we want to be able to roll back if a bad MC caused an unreconcilable state
+		if (dconfig != cconfig || nodeNotReady) && dstate != daemonconsts.MachineConfigDaemonStateUnreconcilable {
```

Comment:
I think this makes sense...just a note to self that the function is now not the inverse of `getReadyMachines`.  

(also this PR will conflict with the doc comments I added in the other PR)

Reply:
> I think this makes sense...just a note to self that the function is now not the inverse of getReadyMachines.

I'm not really sure about this actually, but wanted to check what tests say :(